client/src/java/com/zimbra/client/ZMailbox.java

```diff
@@ -1262,6 +1284,12 @@ public synchronized void handleModify(ZModifyEvent event, ZMailbox mailbox) thro
                 if (contact != null) {
                     contact.modifyNotification(mce);
                 }
+            } else if (event instanceof ZModifySmartFolderEvent) {
+                ZModifySmartFolderEvent msfe = (ZModifySmartFolderEvent) event;
+                ZSmartFolder smartFolder = getSmartFolderById(msfe.getId());
+                if (smartFolder != null) {
```

Comment:
Worth a warning if smartFolder is null?

Reply:
I modeled this after how ZMailbox handles other modification notifications, which don't do anything if the underlying object doesn't exist. Perhaps it's possible for a modify notification to come through for which the item is not cached on the client?